src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -22,6 +22,8 @@ internal readonly partial record struct Checksum
 
     private static readonly ObjectPool<XxHash128> s_incrementalHashPool =
         new(() => new(), size: 20);
+    private static readonly ObjectPool<ObjectWriter> s_objectWriterPool =
+        new(() => new(SerializableBytes.CreateWritableStream(), leaveOpen: true, writeValidationBytes: true), size: 4);
```

Comment:
why size:4?

Reply:
It's because the object is used for such a short period, that it's actually pretty unlikely to have concurrent usage of different items from the pool. I'll add a comment indicating such.